library/core/src/num/mod.rs

```diff
@@ -5006,6 +5007,76 @@ impl u8 {
     pub const fn is_ascii_control(&self) -> bool {
         matches!(*self, b'\0'..=b'\x1F' | b'\x7F')
     }
+
+    /// Returns an iterator that produces an escaped version of a `u8`.
+    ///
+    /// The default is chosen with a bias toward producing literals that are
+    /// legal in a variety of languages, including C++11 and similar C-family
+    /// languages. The exact rules are:
+    ///
+    /// * Tab is escaped as `\t`.
+    /// * Carriage return is escaped as `\r`.
+    /// * Line feed is escaped as `\n`.
+    /// * Single quote is escaped as `\'`.
+    /// * Double quote is escaped as `\"`.
+    /// * Backslash is escaped as `\\`.
+    /// * Any character in the 'printable ASCII' range `0x20` .. `0x7e`
+    ///   inclusive is not escaped.
+    /// * Any other chars are given hex escapes of the form '\xNN'.
+    /// * Unicode escapes are never generated by this function.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #![feature(inherent_ascii_escape)]
+    ///
+    /// use std::ascii;
+    ///
+    /// let escaped = ascii::escape_default(b'0').next().unwrap();
+    /// assert_eq!(b'0', escaped);
+    ///
+    /// let mut escaped = ascii::escape_default(b'\t');
+    ///
+    /// assert_eq!(b'\\', escaped.next().unwrap());
+    /// assert_eq!(b't', escaped.next().unwrap());
+    ///
+    /// let mut escaped = ascii::escape_default(b'\r');
+    ///
+    /// assert_eq!(b'\\', escaped.next().unwrap());
+    /// assert_eq!(b'r', escaped.next().unwrap());
+    ///
+    /// let mut escaped = ascii::escape_default(b'\n');
+    ///
+    /// assert_eq!(b'\\', escaped.next().unwrap());
+    /// assert_eq!(b'n', escaped.next().unwrap());
+    ///
+    /// let mut escaped = ascii::escape_default(b'\'');
+    ///
+    /// assert_eq!(b'\\', escaped.next().unwrap());
+    /// assert_eq!(b'\'', escaped.next().unwrap());
+    ///
+    /// let mut escaped = ascii::escape_default(b'"');
+    ///
+    /// assert_eq!(b'\\', escaped.next().unwrap());
+    /// assert_eq!(b'"', escaped.next().unwrap());
+    ///
+    /// let mut escaped = ascii::escape_default(b'\\');
+    ///
+    /// assert_eq!(b'\\', escaped.next().unwrap());
+    /// assert_eq!(b'\\', escaped.next().unwrap());
+    ///
+    /// let mut escaped = ascii::escape_default(b'\x9d');
+    ///
+    /// assert_eq!(b'\\', escaped.next().unwrap());
+    /// assert_eq!(b'x', escaped.next().unwrap());
+    /// assert_eq!(b'9', escaped.next().unwrap());
+    /// assert_eq!(b'd', escaped.next().unwrap());
+    /// ```
+    #[unstable(feature = "inherent_ascii_escape", issue = "none")]
+    #[inline]
+    pub fn escape_default(&self) -> ascii::EscapeDefault {
```

Comment:
hand-wavy concern: I believe all the `ascii_xxx` methods on `u8` ideally should have had `self` rather than `&self` argument, to be consistent with all other methods. The reason for `&self` I believe is historical, dating back to `AsciiExt` trait. 

Should we use `&self` here to be consistent with other ascii method, or should we use `self`, to be consistent with non-ascii methods, and with the general pattern that small things are passed by `self`?

Reply:
Agree with you, am fine with either convention.

Reply:
Let's use `self` here instead of `&self`. In this case there's not much value in keeping consistency with this historical 'mistake'. (Since `some_byte.escape_ascii()` will work either way.)